microsoft/CBL-Mariner
# Allow DHCP6 related packets through firewall

###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*

---

###### Summary 

These changes are needed for Mariner clients to recieve DHCPv6 packets. This PR heavily borrows from https://github.com/microsoft/CBL-Mariner/pull/5071

As stated in the DHCPv6 Protocol (https://datatracker.ietf.org/doc/html/rfc8415) 
- udp on port 546 is needed for clients
- icmpv6 is needed. In an effort to reduce the changes as much as possible, only add types that are known to be needed. However, unclear if this is actually a worth while effort or if it will to more problems rather than allowing all icmpv6 packets

[NOTES]
- %post section does not seem to actually restart the iptables.service as the new rules are not seen when using `ip6tables -L`. Requires manually restarting the service or restarting the machine
- rerunning the iptables.service is not enough for the new ip local addresses to be seen. Need to also restart networkd

###### Change Log  



- Add rules to accept DHCPv6 packets

###### Does this affect the toolchain?  



**NO**

###### Associated issues  


- https://microsoft.visualstudio.com/OS/_workitems/edit/44612207

###### Test Methodology

- local build and install on Azure VM

Test method:
On fresh Azure Linux VM which supports dual stack (as seen here: https://learn.microsoft.com/en-us/azure/virtual-network/ip-services/create-vm-dual-stack-ipv6-cli#code-try-0)
1. check that ip6tables -L has no icmpv6 rules
2. run tcpdump -n -i eth0 ip6 --> see no dhcp6 solicit or neighbor solicitation
3. run ip addr and see that eth0 has only link local addresses
4. load new rules and restart networkd and iptables
5. run tcpdump -n -i eth0 ip6 --> see dhcp6 solicit and neighbor solicitation messages
6. run ip addr and see that eth0 has additional local addresses
